client/components/login/login.js

```diff
@@ -177,6 +177,7 @@ export default class Login extends React.Component {
       if (!remember_me)
         sessionStorage.setItem(`${orgSlug}_auth_token`, data.key);
       authenticate(true);
+      sessionStorage.setItem(`${orgSlug}_just_authenticated`, true);
```

Comment:
I see that in the same method `componentDidMount` you add this entry to `sessionStorage` and then remove it straightaway, so I wonder why use `sessionStorage` then and not use any other solution based on pure JS?

In some corner cases `sessionStorage` may fail or may not be implemented properly (think about mini browsers, android web views, only God knows on what kind of browsers this will be run), why do we want to depend on something that is optional and can fail if this feature is the key that must not fail?

I recommend avoiding session storage here and use pure JS (either react state, a global variable, whatever you like).

Reply:
I have to use sessionStorage because I want to get the status of just_authenticated from the login component. Another possible solution is using redux but I thought it would be too long for this solution. Should I implement this using redux?